packages/govern-console/src/components/TransactionKeeper/TransactionKeeper.tsx

```diff
@@ -1,64 +1,29 @@
 import React from 'react';
-import {
-  CustomTransaction,
-  CustomTransactionStatus,
-  Response,
-} from 'utils/types';
-import { styled, useTheme } from '@material-ui/core/styles';
-import MUICard from '@material-ui/core/Card';
-import Typography from '@material-ui/core/Typography';
-import TransactionList from 'components/TransactionKeeper/TransactionList';
-import { ANButton } from 'components/Button/ANButton';
+import { CustomTransaction, CustomTransactionStatus } from 'utils/types';
 import produce from 'immer';
+import TransactionKeeperInitial from './TransactionKeeperInitial';
+import TransactionKeeperProgress from './TransactionKeeperProgress';
+import TransactionKeeperSuccess from './TransactionKeeperSuccess';
+import TransactionKeeperFailure from './TransactionKeeperFailure';
 
-//* Styled Components
-const Wrapper = styled(MUICard)(({ theme }) => ({
-  width: '100%',
-  boxSizing: 'border-box',
-  minHeight: '100%',
-  height: 'fit-content',
-  backgroundColor: theme.custom.transactionKeeper.wrapper.background,
-  boxShadow: '0px 13px 9px rgba(180, 193, 228, 0.35)',
-  borderRadius: '16px',
-  padding: '27px 24px 24px 24px',
-}));
-const Title = styled(Typography)(({ theme }) => ({
-  ...theme.custom.transactionKeeper.title,
-  width: '100%',
-  height: 'fit-content',
-  boxSizing: 'border-box',
-  textAlign: 'center',
-  marginBottom: '21px',
-}));
+enum TransactionKeeperState {
+  Initial,
+  Processing,
+  Success,
+  Failure,
+}
 
-const TransactionMessagesCard = styled(MUICard)(({ theme }) => ({
-  ...theme.custom.transactionKeeper.transactionMessagesCard,
-  padding: '16px',
-  marginBottom: '16px',
-}));
-const BoldText = styled(Typography)(({ theme }) => ({
-  ...theme.custom.transactionKeeper.transactionMessagesCard.boldText,
-}));
-const MessageText = styled(Typography)(({ theme }) => ({
-  ...theme.custom.transactionKeeper.transactionMessagesCard.text,
-}));
-const TransactionStatusWrapper = styled('div')({
-  boxSizing: 'border-box',
-  height: 'fit-content',
-  width: '100%',
-  marginBottom: '30px',
-});
-const CloseButton = styled('img')({
-  height: '18px',
-  width: '18px',
-});
-const TransactionListWrapper = styled('ul')(({ theme }) => ({
-  paddingInlineStart: '30px',
-}));
-const TransactionListItem = styled('li')(({ theme }) => ({
-  padding: 0,
-}));
-//* End of Styled Components
+const getErrorFromException = (ex: any): string => {
+  let errorMessage = ex.error?.message || ex.message;
```

Comment:
@yuetloo what do you think if we also add `ex.reason` beside `ex.message` ?

Reply:
sure.  I'll add that.